### PR TITLE
Clean up the crypto parameters and ciphertext

### LINKED ISSUE #143: Clean up the crypto parameters
Remove exra parameters and normalize the names, like depth and maxDepth

### LINKED ISSUE #144: Clean up the ciphertext class
Review the use of depth
Review the use of level

### PR DESCRIPTION
**Pull request by dsuponitskiy**
closes #143 
closes #144

### CODE REVIEW

File: src/pke/examples/advanced-real-numbers.cpp [hunk 1]
```diff
@@ -244,7 +244,7 @@ void AutomaticRescaleDemo(ScalingTechnique scalTech) {
     //  // ones printed above.
     //  std::cout << "Ciphertext c:" << std::endl;
     //  std::cout << "\t scaling factor: " << c->GetScalingFactor() << std::endl;
-    //  std::cout << "\t depth: " << c->GetDepth() << std::endl;
+    //  std::cout << "\t depth: " << c->GetNoiseScaleDeg() << std::endl;
```
Comment (sarojaduality) [inline src/pke/examples/advanced-real-numbers.cpp:247]: Change "\t depth" to "\t noiseScaleDeg"

**Comment by dsuponitskiy** [comment]
Manually started 4 CI jobs

File: src/pke/include/cryptocontext.h [hunk 2]
```diff
@@ -1016,7 +1016,7 @@ class CryptoContextImpl : public Serializable {
             ciphertext->SetEncodingType(plaintext->GetEncodingType());
             ciphertext->SetScalingFactor(plaintext->GetScalingFactor());
             ciphertext->SetScalingFactorInt(plaintext->GetScalingFactorInt());
-            ciphertext->SetDepth(plaintext->GetDepth());
+            ciphertext->SetNoiseScaleDeg(plaintext->GetDepth());
```
Comment (sarojaduality) [inline src/pke/include/cryptocontext.h:1019]: Why are we changing the ciphertext param depth to noiseScaleDeg but not the plaintext param?

Comment (dsuponitskiy) [reply src/pke/include/cryptocontext.h:1019]: yes

**Comment by dsuponitskiy** [comment]
Made all additional changes. The code is ready to be reviewed

**Review by sarojaduality — APPROVED** [review]

**Pull request merged**